queries/sourcepawn/highlights.scm

```diff
@@ -0,0 +1,255 @@
+; Assume all-caps names are constants
+((identifier) @constant
+ (#match? @constant "^[A-Z][A-Z\\d_]+$'"))
+
+; Function definitions/declarations
+(function_definition
+  name: (identifier) @function)
+(function_declaration
+  name: (identifier) @function)
+(parameter_declaration
+  name: (identifier) @variable.parameter)
+
+; Methods / Properties
+(field_access
+  field: (identifier) @field)
+
+; Function calls
+(call_expression
+  function: (identifier) @function)
+(call_expression
+  function: (field_access
+    field: (identifier) @method.call)) ; Must be after field_access
+
+; Types
+(builtin_type) @type.builtin
+(type (identifier) @type)
+(any_type) @type
+
+; Variables
+(variable_storage_class) @storageclass
+(variable_declaration 
+  name: (identifier) @variable)
+(old_variable_declaration
+  name: (identifier) @variable)
+
+; Preprocessor
+(preproc_include) @include
+(preproc_tryinclude) @include
+(system_lib_string) @string
+(string_literal) @string
+
+(preproc_assert) @preproc
+(preproc_pragma) @preproc
+(preproc_arg) @constant
+(preproc_macro) @function.macro
+(macro_param) @parameter
+(preproc_if) @preproc
+(preproc_else) @preproc
+(preproc_elseif) @preproc
+(preproc_endif) @preproc
+(preproc_endinput) @preproc
+(preproc_define) @define
+(preproc_define
+  name: (identifier) @constant)
+(preproc_undefine) @define
+(preproc_undefine
+  name: (identifier) @constant)
+(preproc_error) @function.macro ; Wrong color?
+(preproc_warning) @function.macro ; Wrong color?
+
+; Statements
+(for_statement) @repeat
+(condition_statement) @conditional
+(while_statement) @repeat
+(do_while_statement) @repeat
+(switch_statement) @conditional
+(switch_case) @conditional
+(ternary_expression) @conditional.ternary
+
+; Expressions
+(view_as) @function.builtin
+(sizeof_expression) @function.macro
+(this) @variable.builtin
+
+; https://github.com/alliedmodders/sourcemod/blob/5c0ae11a4619e9cba93478683c7737253ea93ba6/plugins/include/handles.inc#L78
+(hardcoded_symbol) @variable.builtin
+
+; Comments
+(comment) @comment
+
+; General
+(parameter_declaration
+  defaultValue: (identifier) @constant)
+(fixed_dimension) @punctuation.bracket ; the [3] in var[3]
```

Comment:
The `3` should be `@number`.

Reply:
The comment was a leftover and no longer relevant, I removed it